File: PROJETO_5 - WESLEY SANTOS/src/Monitor.py
class Pessoa:
    def __init__(self, nome, email):
        self.nome = nome
        self.email = email

class Aluno(Pessoa):
    def __init__(self, nome, email, matricula, curso):
        super().__init__(nome, email)
        self.matricula = matricula
        self.curso = curso

    def gerar_relatorio(self):
        print(f"Nome: {self.nome}")
        print(f"Email: {self.email}")
        print(f"Matrícula: {self.matricula}")
        print(f"Curso: {self.curso}")

class Professor(Pessoa):
    def __init__(self, nome, email, area_de_atuacao):
        super().__init__(nome, email)
        self.area_de_atuacao = area_de_atuacao

class Monitor(Aluno, Professor):
    def __init__(self, nome, email, matricula, curso, area_de_atuacao):
        Aluno.__init__(self, nome, email, matricula, curso)
        Professor.__init__(self, nome, email, area_de_atuacao)
        self.disciplinas_monitoradas = []

    def adicionar_disciplina_monitorada(self, disciplina):
        self.disciplinas_monitoradas.append(disciplina)

    def gerar_relatorio(self):
        super(Aluno, self).gerar_relatorio()
        print("Disciplinas Monitoradas:")
        for disciplina in self.disciplinas_monitoradas:
            print(f" - {disciplina}")

File: Projeto_5/Projeto_5-Paulo_Victor/src/Monitor.cpp
#include "../include/Monitor.h"

Monitor::Monitor(const std::string& nome, const std::string& email, const std::string& matricula,
                 const std::string& curso, const std::string& areaDeAtuacao)
    : Usuario(nome, email, "Monitor"), Aluno(nome, email, matricula, curso), Professor(nome, email, areaDeAtuacao) {}

void Monitor::adicionarDisciplinaMonitorada(const std::string& disciplina) {
    disciplinasMonitoradas.push_back(disciplina);
}

void Monitor::gerarRelatorio() const {
    Aluno::gerarRelatorio();
    std::cout << "Disciplinas Monitoradas:\n";
    for (const auto& disciplina : disciplinasMonitoradas) {
        std::cout << "- " << disciplina << "\n";
    }
}
